### PR TITLE
Analytics

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Fetch analytics data from API/Google Sheet

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Screenshots
Screenshot 2021-06-30 at 07 36 01

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: public/js/analytics.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+const getAnalytics = async (county) => {
+  const pm1Elem = document.querySelector('.pm1-range .title');
+  const pm2Elem = document.querySelector('.pm2-5-range .title');
+  const pm0Elem = document.querySelector('.pm10-range .title');
+  const noiseElem = document.querySelector('.noise-range .title');
```
Comment (kilemensi) [inline public/js/analytics.js]: These don't change on method call so they can be declared outside the method.

File: public/js/analytics.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  pm1Elem.innerHTML = 0;
+  pm2Elem.innerHTML = '0';
```
Comment (kilemensi) [inline public/js/analytics.js]: Why mixing `'0'` and `0`?

File: public/js/analytics.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+            break;
+          case 'P0':
+            updateMap('pm0', data);
+            break;
+          case 'noise_Leq':
+            updateMap('noise', data);
+            break;
+          default:
+            break;
+        }
```
Comment (kilemensi) [inline public/js/analytics.js]: If you declare rangeMap to use P1, P2, etc., there wouldn't be a need for this switch, just:
```suggestion
        updateMap( data);
```
And `updateMap` would use `data.value_type` to keep track of P1, P2, etc.

File: public/js/analytics.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      });
+    });
+  };
+  const nodes = await window.aq.getNodes();
+  const counties = await window.sheets.getCityCountyMap();
```
Comment (kilemensi) [inline public/js/analytics.js]:
```suggestion
  const cityCountyMap = await window.sheets.getCityCountyMap();
```

File: public/js/analytics.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+  const selectedCounty = nodes.filter((node) => {
```
Comment (kilemensi) [inline public/js/analytics.js]:
```suggestion
  const selectedCountyNodes = nodes.filter((node) => {
```

File: public/js/analytics.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    return counties[node.location.city] === county;
+  });
+
+  const sensorData = selectedCounty
+    ?.map((node) => node.sensors)
+    ?.map((data) => data.map((result) => result.sensordatas));
```
Comment (kilemensi) [inline public/js/analytics.js]:
```suggestion
  const sensorDatas = selectedCounty
    ?.flatMap((node) => node.sensors)
    ?.flatMap((data) => data.map((result) => result.sensordatas));
```

[flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)

File: public/js/init.js [hunk 7]
```diff
@@ -514,6 +516,7 @@ async function init() {
   // TODO(kilemensi): Load other data here
 
   handleLocationChange('nairobi');
+  window.analytics.getAnalytics('Nairobi');
```
Comment (kilemensi) [inline public/js/init.js]: Lets choose one (`slug` or `Name`) and stick to it for all of the `handleLocationChange` methods.

**Review by kilemensi — COMMENTED** [review]
👍🏾

File: public/js/analytics.js (hunk 2 above)
Comment (gertie-sheshe) [reply public/js/analytics.js]: Ah, I was testing something and forgot to revert. Fixed.

File: public/js/analytics.js (hunk 3 above)
Comment (gertie-sheshe) [reply public/js/analytics.js]: You're right. Fixed.

File: public/js/analytics.js (hunk 4 above)
Comment (gertie-sheshe) [reply public/js/analytics.js]: Fixed

File: public/js/analytics.js (hunk 5 above)
Comment (gertie-sheshe) [reply public/js/analytics.js]: Fixed

File: public/js/analytics.js (hunk 1 above)
Comment (gertie-sheshe) [reply public/js/analytics.js]: Fixed

File: public/js/init.js (hunk 7 above)
Comment (gertie-sheshe) [reply public/js/init.js]: Done

File: public/js/analytics.js (hunk 6 above)
Comment (gertie-sheshe) [reply public/js/analytics.js]: Done

File: public/js/init.js [hunk 8]
```diff
@@ -514,6 +516,7 @@ async function init() {
   // TODO(kilemensi): Load other data here
 
   handleLocationChange('nairobi');
+  window.analytics.getAnalytics(countiesLocation.nairobi.name);
```
Comment (kilemensi) [inline public/js/init.js]: Why is this here?

**Review by kilemensi — APPROVED** [review]
🚀

File: public/js/init.js (hunk 8 above)
Comment (gertie-sheshe) [reply public/js/init.js]: Removed
